### DISCUSSION TITLE
[Feature Request] Sub-tasks

### DISCUSSION THREAD

**Discussion by Dyberg** [donetick/donetick#25] [Ideas]
Just started using DoneTick, which is what me and the wife have been looking after for quite some time.
So I just wanted to start by saying thank you for the great work! 

The wife and I was thinking if sub-tasks is on the board? So if for example I create a task saying "Clean the bathroom" I can add sub-tasks "Remove dust", "Vacuum Floor" etc.?

Let me know if you need anything, sadly I can't help write this, just come up with the idea.

**Comment by ados8** [comment]
I'm looking to replace Android Tasks which is a free open source app.
However there's a major gap in the features that Tasks offers such as sub tasks etc.
Would very much like to see some of the features brought into Donetick.

**Comment by meauxt** [comment]
will try to give it some thought and see how to do it without sacrificing the simplicity!

**Reply by ados8** [reply]
Any updates on this?
Right now I have this in a docker to check out the new features after reviewing the release notes but I'm unable to daily the app.
The option to have subtasks is deal breaker and don't know of a task app that doesn't have this feature.

For me I have to be able to separate out the tasks to have things manageable.
Here's some easy examples for needing subtasks.
The most basic, a shopping list, unless you want 30 tasks for each item or dump into a comment box you need separation.
Organising your week by day, Monday, Tuesday etc.
The day is the task and the subtasks are the items, this is how I manage my week and so I regularly go 3 layers down.
Monday is the task, then there's subtasks but many times a subtasks needs subtasks.
Easy example people likely do, visit a business.
In this case Monday is the task, the business is a subtask and there's subtasks for the business for all the things you need to get done.
Have it separate like this makes the workload easier to see and if you run out of time or something changes you can move a subtasks and any subtasks it has to another primary tasks e.g. move the Monday business job to Tuesday.

**Comment by ados8** [comment]
@meauxt I get wanting to keep simplicity but sub tasks I think is universally done, at least every task app I've ever used.
I'm currently moving away from Android Tasks (org.tasks) and it not only supports sub tasks but infinitely, as in sub task in a sub task in...
While that is overkill I think the option to have a shopping list task and sub tasks for all the items to tick off is important.

When implementing subtasks I would highly recommend parent transferring option.
Some task apps forget this and it makes rescheduling sub tasks painful.
Say you have a task list for Monday and get most of them done but not all subtasks or a subtask due date changes.
Some apps have no option but to force you to delete and make a new subtask under another task.
Other more clever apps allow moving/adopting the subtask to a new parent task.

**Comment by ados8** [comment]
Thank you for implementing sub tasks.
Could we please get 1 more level, sub task for sub task.
There's also a small bug when deleting a sub task.
If you make 2 or more sub tasks and don't yet save. If you delete one it will delete all except for those which were previously saved.

**Reply by meauxt** [reply]
You got it! Nested tasks were added as v0.1.37. also the issue you mentioned should be resolved now
 @Dyberg let me know if you like how it working now!

**Reply by ados8** [reply]
Thank you, one of the most active developers and your work is very appreciated.
I see we have infinite subtask levels! I do sometimes go more than 2 levels so this is above and beyond.
Donetick has hit the feature set that allows me to transfer over completely, all other task apps watch out 🥇.

**Comment by lylebrown** [comment]
Would you be open to displaying subtasks under the main task entry? It would be nice to see the first level of subtasks without clicking in.

**Reply by ados8** [reply]
Very much agree on this, I'm used to Tasks.org which you can move are mark off tasks and subtasks without needing to edit the primary task. Since your commonly interacting with subtasks it really does save time.

**Comment by Snuupy** [comment]
is it possible for subtasks to be able to be repeated?

for example for laundry I'll have p1's bedsheets, p2's bedsheets

sometimes they'll be done at the same time (so I want to just check off the parent task "laundry" to check off both the subtasks), but other times they'll be done separately (and I only want to check off 1 of the subtasks but leave the other one with the existing due date)

cc https://discord.com/channels/1272383484509421639/1324236686720237568/1361224077502840933

**Reply by ados8** [reply]
I think that's a bit much, of all the task apps I've used none have allow sub tasks to repeat.
That's what the primary task is for, you can have the sub tasks under that and a note that even week is this and odd week is this etc.

**Reply by Snuupy** [reply]
my workaround right now is just to list subtasks as main tasks and use tags to group them together

**Comment by ados8** [comment]
I would like to mark this one as resolved soon if that's ok with everyone.
However, I noticed immediately as sub sub tasks were implemented that there was a bug but held off raising more issues to give the dev a break 😊.
I wasn't able to migrate over from Android Task.org so before I close this one I would like to get these last bugs squashed.

The bugs, I wasn't able to replicate 1/2 from web but I suspect the Android app is to blame which I've stopped using anyway.
The bug I was getting on Android had sub sub tasks replicate 2+ times.
If for example you had task 1, sub task 1 and sub sub task 1, the latter would also be a sub task in duplication.
Sometimes these duplicated 3+ times and cannot be deleted but as said I wasn't able to replicate from the web for recording.

The other bug I could replicate 100% of the time is you cannot make a sub task and sub sub tasks together, any sub sub tasks are discarded.
The workaround is to make the sub tasks, save, then edit and create your sub sub tasks.

Now that I'm using the web for mobile with webhooks I'll try and migrate over so I can test in high volume.
I would like to use the Android app with potential widgets one day but the app is buggy and doesn't appear to keep up with server updates.

Thank you for all the hard work.

**Comment by Dyberg** [comment]
Thank you for all the hard work, and for actually taking up feature requests :-)
Sorry about the late reply, some personal issues have come in so that I unfortunately can't be that active for now.
Will test asap, but I'm sure everything is working as I can see on other comments :-)

Please mark this request as resolved :-)

**Comment by ados8** [comment]
A very wanted feature for me is NLP with sub tasks.
I know the logistics of that would be crazy so what if the user was just expected to use specific key words to trigger the events.
I think it would also be subtasks only, no sub subtasks to keep complexity down.

Here's an example of how I'm thinking this would work.
`Groceries Saturday at 9 AM subtask milk subtask eggs subtask bread`
The NLP system would work on the first portion and when seeing `subtask` know to make a new subtask for the next series of word until the end or until the next `subtask` keyword and make another and so forth.
Optionally you could also have uncommon special characters that trigger, here's the same example but with that.
`Groceries Saturday at 9 AM * milk * eggs * bread`

This would be great for those that use sub tasks as currently it's chose between NLP or subtasks.

@meauxt

**Comment by viclive79** [comment]
Is there a way to assign the subtasks to different members of the circle. For example right now I want 3 kids to clean their rooms and have to make multiple tasks for each kid assigned to only them. I'm not sure if it would be better if the task was clean rum with a sub task for each kid.  It might make a cleaner experience but not sure if it could do all the tracking.
